pkg/genericapiserver/endpoints/filters/impersonation.go

```diff
@@ -29,8 +29,8 @@ import (
 	"k8s.io/apiserver/pkg/authorization/authorizer"
 	"k8s.io/apiserver/pkg/endpoints/request"
 	"k8s.io/apiserver/pkg/server/httplog"
-	"k8s.io/kubernetes/pkg/api"
-	authenticationapi "k8s.io/kubernetes/pkg/apis/authentication"
+	"k8s.io/client-go/pkg/api"
+	authenticationapi "k8s.io/client-go/pkg/apis/authentication"
```

Comment:
this definitely makes the dependency tree weird.  We should have issue to deal with it.  server shouldn't trust client for its impersonation headers.  I think I'm actually in favor of a copy for the short term instead.